docs/webmaster-modules.md

```diff
@@ -0,0 +1,107 @@
+
+Adding new configuration for modules not in DPL is more safe, but you
+should look out for dependencies. If a configuration depends on a
+particular node type or field being available for instance.
+
+## Overview
+
+Configuration handling in webmaster modules consists of three parts:
+
+1. The configuration itself, in Drupals configuration export YAML
+   format, in a `config/sync` directory in the module.
+2. An event subscriber that overlays the modules configuration when
+   Drupal imports the configuration.
+3. A `hook_install` and possibly `hook_update_N` functions that
+   trigger configuration import when the module is installed and
+   updated.
+
+We'll describe the parts in detail in the following walk-through.
+
+## Walk-through
+
+### Initial configuration
+
+1. Start with a fresh DPL site with the code base from git, that has
+   up to date configuration (`task dev:cli -- drush cim` should do
+   nothing).
+2. Make the required configuration changes in the Drupal
+   administration interface.
+3. Run `task dev:cli -- drush cex -y` to export the configuration.
+4. Now `git status` (or your preferred Git tool) tells you which files
+   has been changed, these are the files you need.
+5. Copy the changed configuration files to `config/sync` in your
+   module and revert the changes to the files in the root
+   `config/sync` folder.
+
+### Event subscriber
+
+In order to get DPL to actually use the configuration you just saved,
+we'll need to make it visible to Drupal. This can be done by
+implementing an event subscriber that overlays the configuration on
+`ConfigEvents::STORAGE_TRANSFORM_IMPORT`. An implementation can be
+found in
+[kdb_brugbyen](https://github.com/kdb/kdb_brugbyen/blob/main/src/EventSubscriber/OverlayConfigEventSubscriber.php).
+You can simply copy that and fix the two references to the module (the
+namespace and the configuration path).
+
+### Install/update hook
+
+The above will add in the module configuration when the configuration
+is imported, but installing or updating a module does not trigger a
+configuration import, so you'll need to do it yourself.
+
+Outside of DPL, triggering a configuration import from install/update
+hooks is not recommended, but inside DPL we're in a controlled
+environment where library sites should always be in sync with the
+provided configuration. So triggering an import to overlay module
+configuration can be considered safe.
+
+Importing configuration is both simple and horribly convoluted. The
+`ConfigImporter` class does all the heavy lifting, but it's not a
+service and requires digging out thirteen different services in order
+to be instantiated. Again, you can just copy the code from
+[kdb_brugbyen](https://github.com/kdb/kdb_brugbyen/blob/main/kdb_brugbyen.install).
+
+## Maintenance
+
+Keeping the module configuration up to date might pose some
+challenges.
+
+Just changing the configuration is often simple, just do the
+modifications, export and copy the changed files.
```

Comment:
Consider explaining whether the developer also needs to ensure that the updated configuration is imported as a part of the change through custom code or should they wait for the next deployment of DPL CMS.